src/contexts/shoppingListsContext.tsx

```diff
@@ -568,7 +693,7 @@ export const ShoppingListsProvider = ({ children }: ProviderProps) => {
     if (authLoading) return
 
     fetchShoppingLists()
-  }, [authLoading, fetchShoppingLists])
+  }, [authLoading, activeGame])
```

Comment:
We had to update the dependency array because `fetchShoppingLists` also updates every time `token` changes, which resulted in a loading screen every time the token was refreshed as a part of this or any API call.